main/mqttDiscovery.cpp

```diff
@@ -593,7 +595,7 @@ void createDiscovery(const char* sensor_type,
       }
     }
 
-    device["via_device"] = String(gateway_name); //device name of the board
+    device["via_device"] = String(gateway_mac); //mac address of the gateway so that the devices link to the gateway
```

Comment:
Could we simplify with : `device["via_device"] = String(getMacAddress());` ?

Reply:
No as then you have the Mac Address of the Bluetooth sensor and not the gateway.

So you need to store it as a global variable that is referenced later.

I thought exactly the same thing before I tested it and found out that wasn't going to work.